src/core/qgsauxiliarystorage.cpp

```diff
@@ -19,13 +19,129 @@
 #include "qgslogger.h"
 #include "qgsslconnect.h"
 #include "qgsproject.h"
+#include "qgspallabeling.h"
+#include "qgsdiagramrenderer.h"
 
 #include <QFile>
 
 const QString AS_JOINFIELD = "ASPK";
 const QString AS_EXTENSION = "qgd";
 const QString AS_JOINPREFIX = "auxiliary_storage_";
 
+QgsAuxiliaryField::QgsAuxiliaryField( const QgsPropertyDefinition &def )
+  : QgsField()
+  , mPropertyDefinition( def )
+{
+  init( def );
+}
+
+QgsAuxiliaryField::QgsAuxiliaryField( const QgsField &f )
+{
+  const QStringList parts = f.name().split( '_' );
```

Comment:
This constructor looks a bit fragile to me - is it required anywhere? Or could we just force use of QgsAuxiliaryField( const QgsPropertyDefinition &def )?

Reply:
Indeed it's pretty fragile, that's why this constructor is private. Only QgsAuxiliaryLayer may use it (friend class).

It's currently used in QgsAuxiliaryLayer::auxiliaryFields method.